doc/sphinx/proof-engine/ltac2.rst

```diff
@@ -126,114 +152,156 @@ Type declarations
 
 One can define new types with the following commands.
 
-.. cmd:: Ltac2 Type {? @ltac2_typeparams } @lident
+.. cmd:: Ltac2 Type {? rec } @tac2typ_def {* with @tac2typ_def }
    :name: Ltac2 Type
 
-   This command defines an abstract type. It has no use for the end user and
-   is dedicated to types representing data coming from the OCaml world.
+   .. insertprodn tac2typ_def tac2rec_field
+
+   .. prodn::
+      tac2typ_def ::= {? @tac2typ_prm } @qualid {? {| := | ::= } @tac2typ_knd }
+      tac2typ_prm ::= @ltac2_typevar
+      | ( {+, @ltac2_typevar } )
+      tac2typ_knd ::= @ltac2_type
+      | [ {? {? %| } {+| @tac2alg_constructor } } ]
+      | [ .. ]
+      | %{ {? {+; @tac2rec_field } {? ; } } %}
+      tac2alg_constructor ::= @ident
+      | @ident ( {*, @ltac2_type } )
+      tac2rec_field ::= {? mutable } @ident : @ltac2_type
+
+   :n:`:=`
+     Defines a type with a manifest
 
-.. cmdv:: Ltac2 Type {? rec} {? @ltac2_typeparams } @lident := @ltac2_typedef
+   :n:`::=`
+     Extends an existing open variant type, a special kind of variant type whose constructors are not
+     statically defined, but can instead be extended dynamically. A typical example
+     is the standard `exn` type for exceptions. Pattern matching on open variants must always
+     include a catch-all clause. They can be extended with this form, in which case
+     :token:`tac2typ_knd` should be in the form :n:`[ {? {? %| } {+| @tac2alg_constructor } } ]`.
 
-   This command defines a type with a manifest. There are four possible
-   kinds of such definitions: alias, variant, record and open variant types.
+     .. todo PR above: does it add an additional constructor for an existing o v type or
+        does it create both the type (and apparently the constructor?
```

Comment:
`Ltac2 Type foo ::= [ Foo ]` gives `Error: Unbound type foo`.  So it only creates the constructor, the type must already exist, as in https://github.com/coq/coq/blob/e6d92a9765f84c80f8c6a102fe5480490c747313/user-contrib/Ltac2/Init.v#L38

Reply:
This seems not yet resolved?  I'm going to unresolve it, but feel free to drop a comment about why it's resolved and re-resolve it.

Reply:
I think the words a few lines above cover it: "Extends an **existing** open variant type"